src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -264,10 +311,12 @@ public Response downloadZip(JsonObject content,
             Result result = controller.execute();
             if (result instanceof Failed)
             {
-               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(result.getMessage()).build();
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(getMessage(result)).build();
             }
             else
             {
+               if (exposedCommands.generateZipCommand(commandName))
```

Comment:
I just used the `generateZipCommand` as being a boolean to return whether or not we should try to create a zip or not. Would you rather the generation of the zip - or the JSON/text  message return was all done in a single method call on the `exposeCommands` object?